### ISSUE TITLE
botsentry covered not anymore

### ISSUE THREAD

**Issue by Malachiel87** [games647/FastLogin#601]
### What happened?

Bot are assaulting my bungee 
https://pastebin.com/QTa7D112
I get these messages and is spammed, a lot of player will lose premium access on my server
As i remember time ago was working perfect.
I am using your latest dev version

### What did you expect?

bot blocked and console cleared

### Steps to reproduce

get a bot attack with botsentry

### Plugin version

1.11-SNAPSHOT-7dd0aa5

### Platform

BungeeCord

### Relevance

- [X] I tried the latest build
- [X] I checked for existing tickets -
If there are, please vote them with a thumps reaction and not create new ones

**Comment by games647** [comment]
I guess what's likely happened is that even legitimate players will be ignored by the simple anti bot limit. This feature will drop all heavy load. This means HTTP as well as database connections. If your database server could withstand the load, you could tune the anti bot configuration.

**Comment by Malachiel87** [comment]
I checked better and i suspect that was changeskin with "server blacklist" feature disabled

**Comment by Malachiel87** [comment]
But there is a way to hide the bot spam? Like
[Fastlogin] Bot attack detected! Protection is on!
and hide these spam of ip on console until it stop
[Fastlogin] Bot attack stopped! Protection is toggled!

**Comment by games647** [comment]
Good point. The current implementation is a leaky bucket (if the bucket is full, requests over it drop until one entry gets free), so it's nothing like on or off. I'll try look into it.

**Comment by Malachiel87** [comment]
I got anothe bot attack, I faced the same issue, what I have to do?

**Comment by games647** [comment]
>you can't make a way to ignore the premium players from bot filter?

It needs to retrieve that data from the database. This limit ignores everything, because there were people complaining about the load on the database, where we can't do anything except don't doing it.

>it should not listen the players outside the botsentry protection,

It's the same issue like in #292, BungeeCord design issue that async tasks don't have an order and execute in parallel.

**Comment by Malachiel87** [comment]
After some analis i noticed the bots where are all whitelisted by our antibot, is that why where hiting fastlogin  and bypassing protection

**Comment by games647** [comment]
I guess whitelisted in your antibot solutions means they are not checked, so yes this likely. However because of the design issue in BungeeCord where both plugins start processing in parallel without an order/dependency it could happen anyway. Except if your anti bot plugin somehow integrates into FastLogin or uses synchronous processing (not recommended). 

Velocity addresses this issue in their recently released 3.0 version with a new API.
